client/src/components/QueryComponent/QueryComponent.js

```diff
@@ -81,17 +66,34 @@ class QueryComponent extends Component {
   }
 
   onChange = evt => {
-    this.setState({ selectedQuery: evt.target.value, selectedQueryKey: null });
+    this.handleEditorChange(evt);
+    this.setState({ selectedQuery: evt, selectedQueryKey: null });
```

Comment:
Is evt an event object? Or is it a string? If it's the latter, then we should probably change the name. If it is an event, then... why is it set to a string in the constructor?

Reply:
It is supposed to be an event but the Monaco Editor exectute the function and send the value as a String so I Change the name to selectedQuery